src/ThreadPool.cc

```diff
@@ -175,6 +175,17 @@ bool ThreadPool::is_shutdown_requested() const
     return shutdown_requested_;
 }
 
+void ThreadPool::set_max_threads(std::size_t num_threads)
+{
+    if (num_threads == 0 || num_threads > max_threads)
```

Comment:
I have not paid much attention to this before. But can we prefix the constant `max_threads` with `ThreadPool::max_threads` to distinguish it from `max_threads_` the class member?